### PR TITLE
Avoid parallel layer downloads in load command

### PR DESCRIPTION
**Pull request by dougm**
docker pull will gracefully wait if there are multiple clients pulling the same image/layers:

```
$ echo redis{,} | xargs -n1 -P2 docker pull
Repository redis already being pulled by another client. Waiting.
...
```

docker load will fail in this case:

```
$ echo *.tgz{,} | xargs -n1 -P2 docker load -i
FATA[0000] Error: rename /var/lib/docker/graph/_tmp/0f6941fb7eaa0a9328fa8a7938bc4cfc869a7914107acf2cd99bc849433824b7 /var/lib/docker/graph/5a762da238b85e305af35f88cbe38d493f893f7dea1b55362bd0d892de53878b: directory not empty
```

This change uses the same mechanism as docker pull to avoid this issue.

### CODE REVIEW

**Comment by jfrazelle** [comment]
ping @dmcgowan @dmp42 @jlhawn

File: graph/load.go [hunk 1]
```diff
@@ -104,6 +104,20 @@ func (s *TagStore) recursiveLoad(eng *engine.Engine, address, tmpImageDir string
...
+		// ensure no two downloads of the same layer happen at the same time
+		if c, err := s.poolAdd("pull", "layer:"+img.ID); err != nil {
+			if c != nil {
+				log.Debugf("Image (id: %s) load is already running, waiting: %v", img.ID, err)
+				<-c
+				return nil
+			} else {
+				return err
+			}
+		} else {
```
Comment (dmcgowan) [inline graph/load.go]: Both these else cases are unnecessary since the if will return

**Comment by dmcgowan** [comment]
small nit then LGTM

**Comment by dougm** [comment]
@dmcgowan fixed, thanks for the review.

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**